### PR TITLE
WT-9441 Fix bound setting in cursor_bound_01 cppsuite stress test

### PR DESCRIPTION
**Pull request by jiechenbo**

### CODE REVIEW

File: test/cppsuite/tests/cursor_bound_01.cpp [hunk 1]
```diff
@@ -49,18 +49,19 @@ class cursor_bound_01 : public test {
     /* Class helper to represent the lower and uppers bounds for the range cursor. */
     class bound {
         public:
-        bound() = default;
-        bound(uint64_t key_size_max, bool lower_bound, std::string key)
-            : _lower_bound(lower_bound), _key(key)
+        bound()
```
Comment (luke-pearson) [inline test/cppsuite/tests/cursor_bound_01.cpp:52]: nice to see the bound class coming in handy! 💪

File: test/cppsuite/tests/cursor_bound_01.cpp [hunk 2]
```diff
@@ -49,18 +49,19 @@ class cursor_bound_01 : public test {
...
+        {
+            this->clear();
```
Comment (etienneptl) [inline test/cppsuite/tests/cursor_bound_01.cpp:54]: I don't think `this` is required

File: test/cppsuite/tests/cursor_bound_01.cpp [hunk 3]
```diff
@@ -229,29 +255,32 @@ class cursor_bound_01 : public test {
...
-                lower_bound = bound(tc->key_size, true, std::string(tc->key_size, 'z'));
-            /* Normal case. */
-            else
-                lower_bound = bound(tc->key_size, true, "0");
+            lower_bound = bound(tc->key_size, true);
             range_cursor->set_key(range_cursor.get(), lower_bound.get_key().c_str());
             ret = range_cursor->bound(range_cursor.get(), lower_bound.get_config().c_str());
             testutil_assert(ret == 0 || ret == EINVAL);
+
+            if (ret == EINVAL)
```
Comment (luke-pearson) [inline test/cppsuite/tests/cursor_bound_01.cpp:263]: Need a comment around why EINVAL is a valid return.

File: test/cppsuite/tests/cursor_bound_01.cpp [hunk 4]
```diff
@@ -578,30 +607,33 @@ class cursor_bound_01 : public test {
...
-            /* Clear all bounds if both bounds don't have a key. */
-            if (bound_pair.first.get_key().empty() && bound_pair.second.get_key().empty()) {
-                lower_bound = bound_pair.first;
-                upper_bound = bound_pair.second;
+            if (!!new_bound_pair.first.get_key().empty())
+                bound_pair.first = new_bound_pair.first;
+            if (!new_bound_pair.second.get_key().empty())
+                bound_pair.second = new_bound_pair.second;
+
+            /* Clear all bounds if both bounds doesn't have a key. */
```
Comment (luke-pearson) [inline test/cppsuite/tests/cursor_bound_01.cpp:633]: Wording: Clear all bounds if both bounds don't have a key.

File: test/cppsuite/tests/cursor_bound_01.cpp [hunk 5]
```diff
@@ -578,30 +607,33 @@ class cursor_bound_01 : public test {
         logger::log_msg(
           LOG_INFO, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
 
-        bound lower_bound, upper_bound;
         std::map<uint64_t, scoped_cursor> cursors;
-
+        std::map<uint64_t, std::pair<bound, bound>> bounds;
```
Comment (luke-pearson) [inline test/cppsuite/tests/cursor_bound_01.cpp:611]: Worth a comment to describe what these maps are and how they relate to one another.

File: test/cppsuite/tests/cursor_bound_01.cpp [hunk 6]
```diff
@@ -49,18 +49,19 @@ class cursor_bound_01 : public test {
...
-            // FIXME: Use random strings, once bounds are implemented properly.
-            // auto key_size =
-            //   random_generator::instance().generate_integer(static_cast<uint64_t>(1),
-            //   key_size_max);
-            // auto random_key = random_generator::instance().generate_random_string(
-            //   key_size, characters_type::ALPHABET);
-            // _key = random_key;
+            auto key_size =
+              random_generator::instance().generate_integer(static_cast<uint64_t>(1), key_size_max);
+            auto random_key = random_generator::instance().generate_random_string(
```
Comment (etienneptl) [inline test/cppsuite/tests/cursor_bound_01.cpp:62]: The variable `random_key` can be removed, we can do this instead:
```
_key = random_generator::instance().generate_random_string(key_size, characters_type::ALPHABET);
```
Same for `set_inclusive`:
```
_inclusive = random_generator::instance().generate_integer(0, 1);
```

**Review by etienneptl — APPROVED** [review]
LGTM, one np

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Review by clarissecheah — APPROVED** [review]
LGTM.

**Pull request merged**